development-guide.md

```diff
@@ -49,6 +49,25 @@ Partners should not add the proto files to their repos. Proto files should be pu
 - Partner code should handle any source/destination errors, retry any transient errors internally without deferring them to Fivetran.
 - Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors instead of throwing exceptions and abruptly closing the connection.
 
+### User Tasks
+
+- Partners can throw alerts on dashboard to notify customers when there is a problem in their Fivetran account, such as a broken connector or an incomplete sync, and tells you how to resolve that problem.
+- Currently we allow only throwing [errors](https://fivetran.com/docs/using-fivetran/fivetran-dashboard/alerts#errors).
+- Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors and pass the message in the following json format to the error description:
+```
+{
+    "message": "Your task message goes here"
+    "alert_type": "TASK"
+}
+```
+
+- Usage example:
+```
+responseObserver.onError(Status.UNAVAILABLE.withDescription(jsonMessage).asException());
+```
+
+- The error description would be converted to a SEVERE log if it is not in the correct JSON format.
```

Comment:
We are not just writing SEVERE log, we are even failing the connector right?

Reply:
Mentioned the same.